### PR TITLE
Add json_string option to chef-client

### PR DESCRIPTION
**Pull request by sylviamoss**
We will have to make a release after this is merged. The components were being registered with the wrong name. 

Closes https://github.com/hashicorp/packer-plugin-chef/issues/8

### CODE REVIEW

File: docs-partials/provisioners/node-attribute-mapping.mdx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  template function.
+
+```hcl
+  json_string = jsonencode({
+    "a" = "b"
+    "foo" = {
+      "bar" = "val"
+      "number" = 1
+    }
+  })
```
Comment (azr) [inline docs-partials/provisioners/node-attribute-mapping.mdx:18]: nice

File: main.go [hunk 2]
```diff
@@ -27,8 +27,8 @@ var (
 
 func main() {
 	pps := plugin.NewSet()
-	pps.RegisterProvisioner("chef-client", new(chefclient.Provisioner))
-	pps.RegisterProvisioner("chef-solo", new(chefsolo.Provisioner))
+	pps.RegisterProvisioner("client", new(chefclient.Provisioner))
+	pps.RegisterProvisioner("solo", new(chefsolo.Provisioner))
```
Comment (azr) [inline main.go:31]: good catch, we probably should delete releases before merging this then.

**Review by azr — APPROVED** [review]
👍🏼 👍🏼 👍🏼 👍🏼 👍🏼

File: main.go (hunk 2 above)
Comment (sylviamoss) [reply main.go:31]: Are you sure? I think this will cause the checksum problem.

Comment (azr) [reply main.go:31]: (because the old binary will need users to use for example the `chef-chef-client` provisioner)

Comment (azr) [reply main.go:31]: (if we delete the tag, the checksums will still exist but packer init won't ever try to install these.)

**Pull request merged**